### DISCUSSION TITLE
Documentation to migrate from @nexus/schema to nexus 1.0

### DISCUSSION THREAD

**Discussion by antoinelin** [graphql-nexus/nexus#733] [Q&A]
Hello Nexus team ! 🖖 

I need to upgrade my Prisma package but I can't use nexus-plugin-prisma without updating @nexus/schema (and so migrate to nexus).
I did not find any documentation to migrate from the previous package to the new and there is a lot of not documented breaking changes (like where has passed the `required` key in `field` ?).
Is there any plan to update the documentation soon ?

Thank you 🙏

**Comment by Epimodev** [comment]
Hello 👋

I'm starting to learn nexus by reading the documentation and it seems some parts aren't updated yet.
For example here: https://nexusjs.org/docs/api/args the import is `@nexus/schema` whereas in the release note here https://github.com/graphql-nexus/nexus/releases/tag/1.0.0 it's written we should download `nexus` package.

So I'm also looking for documentation update.

**Comment by LRipi** [comment]
Hello @toinelin 
You'll find the braking changes right here: https://www.prisma.io/blog/announcing-the-release-of-nexus-schema-v1-b5eno5g08d0b
To answer your question, the `required` key is now set on the declaration of your argument.

Instead of doing:
```
const Post = objectType({
  name: 'Post',
  definition(t) {
    t.id('id', { required: true })
    t.string('title', { required: true })
    t.string('body', { required: false })
  },
})
```
You must do like this:
```
const Post = objectType({
  name: 'Post',
  definition(t) {
    t.nonNull.id('id')
    t.nonNull.string('title')
    t.string('body')
  },
})
```

Hope it help and answer your question 😄

**Comment by antoinelin** [comment]
Hello @LRipi !

Thank you your precise answer!
I took some time to re-read all the documentations checking my need, it seems okay right now.
We agree that the way to add required args on a `Query` type is to add `nonNull` as module imported from `nexus`? :

```
export const Posts = extendType({
  type: 'Query',
  definition(t) {
    t.nonNull.list.field('Posts', {
      type: 'Match',
      args: {
        page: nonNull(intArg()),
        pageSize: nonNull(intArg()),
      },
     ...
   }
  }
})
```

BTW really like nexus, best discovery for 2020 tech wise. Paired with Prisma, it is neat!
